pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -7,23 +8,38 @@ import (
 	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
 	"github.com/rancher/rancher/pkg/controllers/dashboard/chart"
 	"github.com/rancher/rancher/pkg/features"
-	namespace "github.com/rancher/rancher/pkg/namespace"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/namespace"
 	"github.com/rancher/rancher/pkg/settings"
 	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/wrangler/pkg/data"
 	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
 	"github.com/rancher/wrangler/pkg/relatedresource"
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/api/errors"
-	apierror "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 const (
 	repoName         = "rancher-charts"
-	webhookChartName = "rancher-webhook"
 	webhookImage     = "rancher/rancher-webhook"
+	priorityClassKey = "priorityClassName"
 )
 
+var (
+	watchedSettings = map[string]struct{}{
+		settings.RancherWebhookMinVersion.Name: {},
+		settings.RancherWebhookVersion.Name:    {},
+		settings.SystemDefaultRegistry.Name:    {},
+		settings.ShellImage.Name:               {},
+	}
+	watchedFeatures = map[string]struct{}{
+		features.MCM.Name():                {},
+		features.EmbeddedClusterAPI.Name(): {},
+	}
```

Comment:
This is a non-trivial departure from the current functionality in at least two ways:
- First, we are enqueing on new settings (system default registry/shell image) which we weren't enqueing on before. I think that this is ok because we use these items to configure/install the webhook, so not enqueing on them in the past could be considered a bug, but it's a change worth noting.
- Second, we are only enqueing on 2 features, whereas before we enqueued on all features. I don't think we can ship this - other controllers for ClusterRepos could be relying on this enqueing functionality (they shouldn't be, but could be) and we can't remove it here. In addition, this isn't related to the new functionality that you are introducing - so it represents an unnecessary risk. 

I think we can justify changing the first item, but not the second.